yktrace/src/tir.rs

```diff
@@ -285,74 +258,143 @@ impl TirTrace {
             }
         }
 
+        // Remove the remnants of the `start_tracing` (1 instruction) and `stop_tracing` (5
+        // instructions) calls at the beginning and end of the trace.  To be sure we don't remove
+        // something important, let's put some asserts in to check these instructions are always
+        // roughly the same.
+        match ops.pop() {
+            Some(TirOp::Statement(Statement::Enter(CallOperand::Fn(s),_,_,_))) => {
+                assert!(s.contains("stop_tracing"))
+            }
+            e => panic!("Expected call to `stop_tracing` here, instead got {:?}.", e)
+        }
+        match ops.pop() {
+            Some(TirOp::Statement(Statement::Assign(_, _))) => {
+            }
+            e => panic!("Expected `Assign` here, instead got {:?}.", e)
+        }
+        match ops.pop() {
+            Some(TirOp::Statement(Statement::Assign(_, Rvalue::Use(Operand::Constant(Constant::Bool(false)))))) => {
+            }
+            e => panic!("Expected `Assign(_, false)` here, instead got {:?}.", e)
+        }
+        for _ in (0..3) {
+            match ops.pop() {
+                Some(TirOp::Statement(Statement::StorageLive(_))) => {
+                }
+                e => panic!("Expected `StorageLive` here, instead got {:?}.", e)
+            }
+        }
+        match ops.remove(0) {
+            TirOp::Statement(Statement::StorageDead(_)) => {
+            }
+            e => panic!("Expected `StorageDead` here, instead got {:?}.", e)
+        }
+
         Ok(Self { ops })
     }
 
-    fn rename_rvalue(rename_map: &HashMap<Local, Vec<Local>>, rvalue: &Rvalue) -> Rvalue {
+    /// Return the TIR operation at index `idx` in the trace.
+    /// The index must not be out of bounds.
+    pub fn op(&self, idx: usize) -> &TirOp {
+        debug_assert!(idx <= self.ops.len() - 1, "bogus trace index");
+        unsafe { &self.ops.get_unchecked(idx) }
+    }
+
+    /// Return the length of the trace measure in operations.
+    pub fn len(&self) -> usize {
+        self.ops.len()
+    }
+}
+
+struct SSA {
```

Comment:
Does this need to be revised in light of our IR not quite being SSA? This only does renamings, but doesn't make new inflections of variable upon mutations I think?

Reply:
I'm open to suggestions. I don't have any.

Reply:
`VarRenamer`?

Reply:
Fixed in 48ca04d